x/ibc-hooks/wasm_hook.go

```diff
@@ -66,13 +68,22 @@ func (h WasmHooks) OnRecvPacketOverride(im IBCMiddleware, ctx sdk.Context, packe
 		return osmoutils.NewStringErrorAcknowledgement("error in wasmhook message validation")
 	}
 
-	// The funds sent on this packet need to be transferred to the wasm hooks module address/
-	// For this, we override the ICS20 packet's Receiver (essentially hijacking the funds for the module)
+	// Calculate the receiver / contract caller based on the packet's channel and sender
+	senderStr := fmt.Sprintf("%s/%s", packet.GetDestChannel(), data.GetSender())
+	senderHash32 := sha256.Sum256([]byte(senderStr))
```

Comment:
We need to ensure this is properly namespaced away from any possible user address, or module account address.

Lets do so, by not hashing directly like this, but by instead using https://github.com/cosmos/cosmos-sdk/blob/main/types/address/hash.go

We get the safety guarantee we need, by just using some `type` field we think won't get re-used, perhaps `ibc-memo-action` or smth